build/Makefile

```diff
@@ -0,0 +1,43 @@
+.PHONY: all clean images
+.DEFAULT_GOAL := all
+
+# Boiler plate for bulding Docker containers.
+# All this must go at top of file I'm afraid.
+IMAGE_PREFIX := weaveworks/build-
+IMAGE_TAG := $(shell ../image-tag)
+UPTODATE := .uptodate
+
+# Building Docker images is now automated. The convention is every directory
+# with a Dockerfile in it builds an image calls quay.io/weaveworks/<dirname>.
+# Dependencies (i.e. things that go in the image) still need to be explicitly
+# declared.
```

Comment:
I realise this is copy-and-pasted, but…

* First sentence is redundant. Please delete.
* "The convention is" also redundant. Please delete.
* "an image calls" -> "an image called"
* It's *not* called `quay.io/weaveworks/<dirname>`, it's called `$(IMAGE_PREFIX)/<dirname>` (I think)